clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientCredentialsProvider.java

```diff
@@ -25,25 +25,32 @@
 import java.io.File;
 import java.io.IOException;
 
-public class BearerTokenCredentialsProvider implements CredentialsProvider {
+public class ZeebeClientCredentialsProvider implements CredentialsProvider {
+
+  public static final String INVALID_PATH_ERROR_MSG =
+      "Expected valid path to Zeebe credentials but '%s' is either invalid or does not point to a file.";
+  private static final TypeReference<ZeebeClientAuthInfo> TYPE_REF =
+      new TypeReference<ZeebeClientAuthInfo>() {};
+  private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
   private static final Key<String> HEADER_AUTH_KEY =
       Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
-  private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
   private final String zeebeCredentialsPath;
 
-  public BearerTokenCredentialsProvider(String zeebeCredentialsPath) {
-    if (zeebeCredentialsPath == null
-        || zeebeCredentialsPath.isEmpty()
-        || !new File(zeebeCredentialsPath).exists()) {
+  public ZeebeClientCredentialsProvider(String zeebeCredentialsPath) {
+    if (isValidCredentialsPath(zeebeCredentialsPath)) {
       throw new IllegalArgumentException(
-          String.format(
-              "Expected valid path to Zeebe credentials but '%s' is either invalid or does not point to a file.",
-              zeebeCredentialsPath));
+          String.format(INVALID_PATH_ERROR_MSG, zeebeCredentialsPath));
     }
 
     this.zeebeCredentialsPath = zeebeCredentialsPath;
   }
 
+  private boolean isValidCredentialsPath(String zeebeCredentialsPath) {
+    return zeebeCredentialsPath == null
+        || zeebeCredentialsPath.isEmpty()
+        || !new File(zeebeCredentialsPath).exists();
```

Comment:
Can we still not create an object in the condition? It makes it more compact but it not readable 😅